core/src/execution/datafusion/spark_hash.rs

```diff
@@ -422,15 +421,16 @@ mod tests {
             Some(0.0),
             Some(-0.0),
             Some(-1.0),
+            None,
```

Comment:
Hmm, because you add null value, the test only tests the logic of null input. Can you have it test both non-null and null paths?